packages/styled-base/src/utils.js

```diff
@@ -1,9 +1,32 @@
 // @flow
 import * as React from 'react'
+import type { ElementType } from 'react'
 import isPropValid from '@emotion/is-prop-valid'
 
 export type Interpolations = Array<any>
 
+export type StyledOptions = {
+  label?: string,
+  shouldForwardProp?: string => boolean,
+  target?: string
+}
+
+export type StyledComponent<P> = React$StatelessFunctionalComponent<P> & {
+  defaultProps: P,
```

Comment:
is this correct? shouldnt `Partial<P>` equivalent from flow be used here?

Reply:
Yeah probably this should be `$Shape<P>`

Reply:
I wasn't unable to use `$Shape<P>` because that would then get assigned to `P` and throw a type error.

It looks like Flow doesn't support `defaultProps`, I typed it as `any` for now :-(

https://github.com/facebook/flow/issues/1660